engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/transform/BpmnResourceTransformer.java

```diff
@@ -60,24 +60,35 @@ public BpmnResourceTransformer(
   @Override
   public Either<Failure, Void> transformResource(
       final DeploymentResource resource, final DeploymentRecord deployment) {
-
-    final BpmnModelInstance definition = readProcessDefinition(resource);
-    final String validationError = validator.validate(definition);
-
-    if (validationError == null) {
-      // transform the model to avoid unexpected failures that are not covered by the validator
-      bpmnTransformer.transformDefinitions(definition);
-
-      return checkForDuplicateBpmnId(definition, resource, deployment)
-          .map(
-              ok -> {
-                transformProcessResource(deployment, resource, definition);
-                return null;
-              });
-
-    } else {
-      final var failureMessage =
-          String.format("'%s': %s", resource.getResourceName(), validationError);
+    try {
+      final BpmnModelInstance definition = readProcessDefinition(resource);
+      final String validationError = validator.validate(definition);
+
+      if (validationError == null) {
+        // transform the model to avoid unexpected failures that are not covered by the validator
+        bpmnTransformer.transformDefinitions(definition);
+
+        return checkForDuplicateBpmnId(definition, resource, deployment)
+            .map(
+                ok -> {
+                  transformProcessResource(deployment, resource, definition);
+                  return null;
+                });
+
+      } else {
+        final var failureMessage =
+            String.format("'%s': %s", resource.getResourceName(), validationError);
+        return Either.left(new Failure(failureMessage));
+      }
+    } catch (final Exception e) {
+      final var cause = e.getCause();
+      final String failureMessage;
+      if (cause instanceof SAXException) {
+        failureMessage =
+            String.format("'%s': SAXException %s", resource.getResourceName(), cause.getMessage());
+      } else {
+        failureMessage = String.format("'%s': %s", resource.getResourceName(), e.getMessage());
+      }
```

Comment:
❌ I think this too much of a catch-all. I would propose that instead we isolate the part which can throw the exception. This is already isolated as it's thrown by the `readProcessDefinition` method. 

We can change the signature of this method and have it return an `Either<Failure, BpmnModelInstance>`. Inside of the method we need to catch a `ModelParseException` and if we do we return an `Either.left` with the failure containing nice message for users. If no exception is thrown we will return an `Either.right` with the model instance.

Reply:
Done.

Hi @remcowesterhoud. Please check this out. :bow: